superset-frontend/src/SqlLab/components/QueryTable/index.tsx

```diff
@@ -76,17 +76,22 @@ const QueryTable = ({
   const theme = useTheme();
   const dispatch = useDispatch();
 
-  const setHeaders = (column: string) => {
-    if (column === 'sql') {
-      return column.toUpperCase();
-    }
-    return column.charAt(0).toUpperCase().concat(column.slice(1));
+  const QUERY_HISTORY_TABLE_HEADERS_LOCALIZED = {
+    state: t('State'),
+    started: t('Started'),
+    duration: t('Duration'),
+    progress: t('Progress'),
+    rows: t('Rows'),
+    sql: t('SQL'),
+    results: t('Results'),
+    actions: t('Actions'),
   };
+
   const columnsOfTable = useMemo(
     () =>
       columns.map(column => ({
         accessor: column,
-        Header: () => setHeaders(column),
+        Header: QUERY_HISTORY_TABLE_HEADERS_LOCALIZED[column],
```

Comment:
This is the only part that makes me a little nervous. I haven't done any real spelunking in the code here, but are we confident that these are the only header strings we'll need to worry about for the foreseeable future?

I wonder if it's safer to leave the function in place as a backup, i.e.

```suggestion
        Header: QUERY_HISTORY_TABLE_HEADERS_LOCALIZED[column] || setHeaders(column),
```